src/group_theory/coset.lean

```diff
@@ -376,6 +376,52 @@ calc α ≃ Σ L : quotient s, {x : α // (x : quotient s) = L} :
     ... ≃ quotient s × s :
   equiv.sigma_equiv_prod _ _
 
+variables {t : subgroup α}
+
+/-- If `H ≤ K` then `G/H ≃ G/K × K/H` -/
+@[simps] def quotient_equiv_prod_of_le' (h_le : s ≤ t)
+  (f : quotient t → α) (hf : function.right_inverse f quotient_group.mk) :
+  quotient s ≃ quotient t × quotient (s.subgroup_of t) :=
+{ to_fun := λ a, ⟨a.map' id (by exact λ b c h, h_le h),
+    a.map' (λ g : α, ⟨(f (quotient.mk' g))⁻¹ * g, quotient.exact' (hf g)⟩) (λ b c h, by
+    { change ((f b)⁻¹ * b)⁻¹ * ((f c)⁻¹ * c) ∈ s,
+      have key : f b = f c := congr_arg f (quotient.sound' (h_le h)),
+      rwa [key, mul_inv_rev, inv_inv, mul_assoc, mul_inv_cancel_left] })⟩,
+  inv_fun := λ a, a.2.map' (λ b, f a.1 * b) (λ b c h, by
+  { change (f a.1 * b)⁻¹ * (f a.1 * c) ∈ s,
+    rwa [mul_inv_rev, mul_assoc, inv_mul_cancel_left] }),
+  left_inv := by
+  { refine quotient.ind' (λ a, _),
+    simp_rw [quotient.map'_mk', id.def, t.coe_mk, mul_inv_cancel_left] },
+  right_inv := by
+  { refine prod.rec _,
+    refine quotient.ind' (λ a, _),
+    refine quotient.ind' (λ b, _),
+    have key : quotient.mk' (f (quotient.mk' a) * b) = quotient.mk' a :=
+    (quotient_group.mk_mul_of_mem (f a) ↑b b.2).trans (hf a),
+    simp_rw [quotient.map'_mk', id.def, key, inv_mul_cancel_left, subtype.coe_eta] } }
+
+/-- If `H ≤ K` then `G/H ≃ G/K × K/H` -/
+@[simps] noncomputable def quotient_equiv_prod_of_le (h_le : s ≤ t) :
+  quotient s ≃ quotient t × quotient (s.subgroup_of t) :=
+quotient_equiv_prod_of_le' h_le quotient.out' quotient.out_eq'
+
+noncomputable def quotient_bot : quotient (⊥ : subgroup α) ≃ α :=
```

Comment:
Do you want to make this a `mul_equiv`? Also, I'm surprised this isn't there yet. It should certainly go to another file.

Reply:
Isn't this the correct file, because this is where quotients are defined?

Reply:
Regarding `mul_equiv`, I think the group structure is proven in a later file (maybe `group_theory/quotient.lean`?) What's the best strategy here?

I could just use the old proof of `group_equiv_quotient_times_subgroup`, and move `quotient_bot` to the later file.

Reply:
Ooh, sorry, you are completely right. I somehow thought I was looking at another file :facepalm:

Reply:
Ok, I've made some changes.